endpoints/s3/src/putobject.cpp

```diff
@@ -58,64 +98,135 @@ void manually_parse_chunked_body_write_to_irods_in_background(
 	parsing_state current_state,
 	size_t chunk_size,
 	const std::string& parsing_buffer_string,
+	bool know_part_offset,
+	bool keep_dstream_open_flag,
 	const std::string func);
 
-/*std::make_shared<incremental_async_read>(
-    parser, session_ptr, response, path, upload_part, upload_part_filename)
-        ->start();*/
 class incremental_async_read : public std::enable_shared_from_this<incremental_async_read>
 {
 	irods::http::session_pointer_type session_ptr_;
 	beast::http::response<beast::http::empty_body> resp_;
 	std::shared_ptr<beast::http::request_parser<boost::beast::http::buffer_body>> parser_;
 	std::string irods_path_;
 	bool upload_part_flag_;
+	bool know_part_offset_flag_;
+	std::size_t part_offset_;
+	std::string upload_id_;
 	std::string part_filename_;
 	std::ofstream part_file_;
-	irods::experimental::io::odstream odstream_;
 	std::vector<char> buffer_;
 	std::size_t total_bytes_read_{};
+	bool keep_dstream_open_flag;
 	std::shared_ptr<irods::experimental::client_connection> conn_;
-	irods::experimental::io::client::default_transport tp_;
+	std::shared_ptr<irods::experimental::io::client::default_transport> tp_;
+	std::shared_ptr<irods::experimental::io::odstream> odstream_;
 
   public:
+	~incremental_async_read()
+	{
+		if (keep_dstream_open_flag) {
+			//std::this_thread::sleep_for(std::chrono::seconds(3));
+			logging::debug("Destructor iRODS called for object that did first open");
```

Comment:
I think if this is just for debugging, it might be better to just remove the destructor. If it seems like it will be really useful, consider surrounding this with a try/catch(...) so that no exceptions escape. Here's what cppreference says (with my own emphasis)... https://en.cppreference.com/w/cpp/language/destructor

> Although [std::uncaught_exceptions](https://en.cppreference.com/w/cpp/error/uncaught_exception) may sometimes be used to detect stack unwinding in progress, **it is generally considered bad practice to allow *any* destructor to terminate by throwing an exception**.

Reply:
I am going to remove it.  I didn't intend on keeping this in.

Reply:
done